cocos/renderer/CCTexture2D.cpp

```diff
@@ -1444,4 +1464,9 @@ void Texture2D::removeSpriteFrameCapInset(SpriteFrame* spriteFrame)
     }
 }
 
+void Texture2D::setAutoGenerateMipmap(bool generateMipmaps)
+{
+    _generateMipmaps = generateMipmaps;
```

Comment:
setAutoGenerateMipmap only mark for initWithMipmaps, need generate mipmap.
initWithMipmaps set _hasMipmaps if generate mipmap.